src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpServer.cs

```diff
@@ -55,15 +60,50 @@ public Task StartAsync<TContext>(IHttpApplication<TContext> application, Cancell
 
         public Task StopAsync(CancellationToken cancellationToken)
         {
-            // TODO: Drain pending requests
+            // First call back into native saying "DON'T SEND ME ANY MORE REQUESTS"
+            void RegisterCancelation()
+            {
+                cancellationToken.Register(() =>
+                {
+                    _shutdownSignal.TrySetResult(null);
```

Comment:
Does this need to call `lNativeMethods.http_stop_calls_into_mananged()`? Or will that be handled somewhere else in the timeout scenario?